### PR TITLE
Add theme styles from designs

### PR DESCRIPTION
**Pull request by alexelash**
## Overview
This PR adds correct styles for font sizes, colors, line-heights and more. I also extended `<Heading>` and `<Text>` and created variants based on the designs for us to use. More details on that in the notes.

Connects #48, connects #47

### Demo
Screen Shot 2019-05-07 at 5 52 20 PM
Screen Shot 2019-05-08 at 11 05 48 AM

Screen Shot 2019-05-07 at 5 52 27 PM
Screen Shot 2019-05-07 at 5 52 29 PM
Screen Shot 2019-05-07 at 5 52 31 PM

### Notes
- `<Text>` and `<Header>` are the first components from Rebass that we are extending. I focused primarily on creating styles for text that exists.
- To style `<Header>` and `<Text>` components in a consistent way, we are using custom variants. Knowing that we will want to have access to theme variables from `theme.js` meant that we cannot use the default variant `textStyles:[]` from Styled System (described [here](https://styled-system.com/api) under "Variant"). The solution I went with allows us to simply pass the prop `variant='variant-type'` to each of these components. We will likely handle `<Button>` variants this way for the same reasons.
- Naming conventions of small, base and medium don't map directly to a font size, they are intentionally scoped to actual usage of each component.
- We will eventually want to change up colors directly within components, but I'm not sure how to set up these variants to be override-able defaults. 
- I added some placeholder background colors to different screens, so that the text is readable. 
- I applied the new extended components to all available screens.
- Note that `<Button>` looks messed up now. I figured I'd hold off on addressing this until #49

## Testing Instructions
* `git pull`
* Switch out the `<Header>` variants to see what they look like
* Switch out the `<Text>` variants to see what they look like.
* Change out a variable within one of the variants (color is an easy one to see).

### CODE REVIEW

File: src/app/src/About.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ import { func, number } from 'prop-types';
 import React from 'react';
 import { connect } from 'react-redux';
 import styled from 'styled-components';
+import { themeGet } from 'styled-system';
+import { Heading } from '../src';
```
Comment (fungjj92) [inline src/app/src/About.js]: `import Heading from './Heading'`?

File: src/app/src/MeetTheFish.js [hunk 2]
```diff
@@ -1,20 +1,21 @@
 import React from 'react';
 import styled from 'styled-components';
-import { Flex, Box, Heading, Text } from 'rebass';
+import { Flex, Box } from 'rebass';
+import { Heading, Text } from '../src';
```
Comment (fungjj92) [inline src/app/src/MeetTheFish.js]: I think we would benefit by having a single file from where the custom styled components can be imported (similar to `main.css`'s role) which is more clear than generally importing from a folder (`../src`). That, or import them individually like `import Heading from './Heading'`

**Review by fungjj92 — COMMENTED** [review]
Looking good! The app is taking shape. Just a comment about imports organization.

File: src/app/src/MeetTheFish.js (hunk 2 above)
Comment (alexelash) [reply src/app/src/MeetTheFish.js]: That makes a ton of sense! I'll take a stab at that instead. Blaster does something similar to what you're describing.

Comment (alexelash) [reply src/app/src/MeetTheFish.js]: What do you think would be a good name for that file?

Comment (fungjj92) [reply src/app/src/MeetTheFish.js]: Great! Hm `custom-styled-components.js`?

**Comment by alexelash** [comment]
@fungjj92 Okay, this is ready for another look! 👀

**Review by fungjj92 — APPROVED** [review]
👍

**Pull request merged**